Panic on trailing invalid key

**Describe the bug**
Given document of `0.`, `go-toml` panics:
```
$ echo -n "0." | gotoml-test-decoder
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/pelletier/go-toml/v2.(*parser).parseSimpleKey(0x200, {0xc0000c6002, 0x1, 0x200})
        /x/src/github.com/pelletier/go-toml/parser.go:700 +0x1cd
github.com/pelletier/go-toml/v2.(*parser).parseKey(0xc0000c8000, {0xc0000c6000, 0xc0000001a0, 0x6bad9aecaea8})
        /x/src/github.com/pelletier/go-toml/parser.go:677 +0x2f7
github.com/pelletier/go-toml/v2.(*parser).parseKeyval(0xc0000c8000, {0xc0000c6000, 0x2, 0x200})
        /x/src/github.com/pelletier/go-toml/parser.go:208 +0x17c
github.com/pelletier/go-toml/v2.(*parser).parseExpression(0xc0000ca000, {0xc0000c6000, 0x40, 0x51e2a0})
        /x/src/github.com/pelletier/go-toml/parser.go:122 +0xa7
github.com/pelletier/go-toml/v2.(*parser).NextExpression(0xc0000c8000)
        /x/src/github.com/pelletier/go-toml/parser.go:62 +0x119
github.com/pelletier/go-toml/v2.(*decoder).nextExpr(...)
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:168
github.com/pelletier/go-toml/v2.(*decoder).fromParser(0xc0000ca000, {0x4e1d80, 0xc0000b4018, 0x100010000488a26})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:222 +0x65
github.com/pelletier/go-toml/v2.(*decoder).FromParser(0xc0000ca000, {0x4da800, 0xc0000b4018})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:208 +0x1e7
github.com/pelletier/go-toml/v2.(*Decoder).Decode(0xc00009aec8, {0x4da800, 0xc0000b4018})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:112 +0x1d2
github.com/pelletier/go-toml/v2/testsuite.DecodeStdin()
        /x/src/github.com/pelletier/go-toml/testsuite/testsuite.go:39 +0x7c
main.main()
        /x/src/github.com/pelletier/go-toml/cmd/gotoml-test-decoder/main.go:20 +0xb9
```

**To Reproduce**
https://go.dev/play/p/nTrQFPp1mOH
```go
toml.Unmarshal([]byte("0."), &v)
```

**Expected behavior**
Not a panic.  Expected error for invalid input.

**Versions**
 - go-toml: d58eb50ebfb929b84ef2af730b2f8503aa4a25fb
 - go version go1.17.5 linux/amd64

**Additional context**
Found while fuzzing (but not differential fuzzing this time).

--- a/parser.go
+++ b/parser.go
@@ -693,6 +693,10 @@ func (p *parser) parseKey(b []byte) (ast.Reference, []byte, error) {
 }
 
 func (p *parser) parseSimpleKey(b []byte) (raw, key, rest []byte, err error) {
+	if len(b) == 0 {
+		return nil, nil, nil, newDecodeError(b, "expected key but found none")
+	}
+
 	// simple-key = quoted-key / unquoted-key
 	// unquoted-key = 1*( ALPHA / DIGIT / %x2D / %x5F ) ; A-Z / a-z / 0-9 / - / _
 	// quoted-key = basic-string / literal-string

--- a/unmarshaler_test.go
+++ b/unmarshaler_test.go
@@ -2326,6 +2326,15 @@ func TestIssue710(t *testing.T) {
 	require.Equal(t, map[string]toml.LocalTime{"0": {Nanosecond: 111111111, Precision: 9}}, v2)
 }
 
+func TestIssue714(t *testing.T) {
+	var v interface{}
+	err := toml.Unmarshal([]byte("0."), &v)
+	require.Error(t, err)
+
+	err = toml.Unmarshal([]byte("0={0=0,"), &v)
+	require.Error(t, err)
+}
+
 func TestUnmarshalDecodeErrors(t *testing.T) {
 	examples := []struct {
 		desc string

